tests/unit/cocoon/test_utilities.py

```diff
@@ -1587,24 +1587,44 @@ def test_identify_cash_items_failed(
                     }
                 },
                 False,
-                ["GBP_EXP", "GBP_EXP", "USD_EXP", "USD_EXP"],
-            ),
-            (
-                "combined_currency_code_inference",
-                {
-                    "cash_identifiers": {
-                        "instrument_name": {
-                            "inst1": "GBP_EXP",
-                            "inst2": "GBP_EXP",
-                            "inst3": "USD_EXP",
-                            "inst4": "",
-                        },
-                    },
-                    "implicit": "internal_currency",
-                },
-                False,
-                ["GBP_EXP", "GBP_EXP", "USD_EXP", "USD_IMP"],
-            ),
+                ["GBP_EXP", "GBP_EXP", "USD_EXP", "USD_EXP", None],
+            )
+        ]
+    )
+    def test_identify_cash_items_remove_false(self, _, cash_flag, remove_cash_items, ground_truth):
+        data = {
+            "instrument_name": ["inst1", "inst2", "inst3", "inst4", "inst5"],
+            "internal_currency": ["GBP_IMP", "GBP_IMP", "USD_IMP", "USD_IMP", "APPLUK"],
+            "Figi": ["BBG01", None, None, None, "BBG02"],
+        }
+        file_type = "instruments"
+        identifier_mapping = {"Figi": "figi"}
+
+        mappings = {
+            file_type: {"identifier_mapping": identifier_mapping,},
+            "cash_flag": cash_flag,
+        }
+        mappings_ground_truth = copy.deepcopy(mappings)
+
+
+        mappings_ground_truth[file_type]["identifier_mapping"][
+            "Currency"
+        ] = "__currency_identifier_for_LUSID"
+
+        dataframe = pd.DataFrame(data)
+
+        dataframe, mappings_test = identify_cash_items(
+            dataframe, mappings, file_type, remove_cash_items
+        )
+
+        self.assertEqual(
```

Comment:
might be better to use `assertListEqual`

Reply:
Made change, also used assertDictEquals